pkg/vsphere/guestinfo/encode.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guestinfo
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	// DefaultTagName value
+	DefaultTagName = "vic"
+	// DefaultPrefix value
+	DefaultPrefix = ""
+)
+
+const (
+	// Invalid value
+	Invalid = 1 << iota
+	// Hidden value
+	Hidden
+	// ReadOnly value
+	ReadOnly
+	// ReadWrite value
+	ReadWrite
+	// NonPersistent value
+	NonPersistent
+	// Volatile value
+	Volatile
+)
+
+// ToString converts a basic type to string
+func ToString(field reflect.Value) string {
+	switch field.Kind() {
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return strconv.FormatInt(field.Int(), 10)
+	case reflect.Bool:
+		return strconv.FormatBool(field.Bool())
+	case reflect.String:
+		return field.String()
+	case reflect.Float32, reflect.Float64:
+		return strconv.FormatFloat(field.Float(), 'E', -1, 64)
+	default:
+		return ""
```

Comment:
We could add time - I'm not sure what configuration element requires it currently unless creation/poweron time isn't part of the standard vmodl. I'd suggest adding things only as we need them for now.